roachtest: more ground work to move tests into `tests`

tbg opened the pull request:
See individual commits.

tbg committed:
roachtest: avoid access to global `local` from tests

Use `c.IsLocal()` instead.

Release note: None

tbg committed:
roachtest: move PredecessorVersions to `tests`

Avoids import conflict as more roachtests move to `tests`.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/67283)
<!-- Reviewable:end -->

tbg committed:
roachtest: fix stray newMonitor

Release note: None

tbg committed:
roachtest: move workerAction to options

Avoids another reference to the main package when tests are moved to a
subpackage.

Release note: None

tbg committed:
roachtest: move tests helpers to `tests/`

Release note: None

tbg committed:
roachtest: make perfArtifactsDir available on `test.Test`

Release note: None

tbg committed:
roachtest: use t.PerfArtifactsDir() instead of perfArtifactsDir

Release note: None

tbg committed:
roachtest: fix (another) stray newMonitor

Release note: None

tbg committed:
roachtest: move monitorImpl to own file

Release note: None

tbg committed:
roachtest: move getDiskUsageInBytes to own file

That way, when the tests are moved wholesale, this method
can move with them, simplifying the process without the
need to export this method in the interim.

Release note: None

tbg committed:
roachtest: move withCommandDetails to `cluster`

This isn't my favorite home for this, but it does need to be moved out
of the `main` package so that tests that use it can move to `tests/`.
It will do for now; better than introducing the inevitable `util`
package too early.

Release note: None

tbg committed:
roachtest: remove `t` from `(Cluster).NewMonitor`

This makes sure that `cluster` does not depend on `test`, and
also makes `NewMonitor` consistent with the remainder of the
interface (which generally hides the `t` in the impl).

Release note: None

tbg committed:
roachtest: expand a TODO about a refactor

Release note: None

tbg committed:
roachtest: don't depend on global `cloud`

Release note: None

erikgrinaker commented on pkg/cmd/roachtest/option/options.go:
```diff
@@ -39,3 +39,19 @@ var StartArgsDontEncrypt = StartArgs("--encrypt=false")
+// Option implements Option.
+func (o WorkerAction) Option() {}
+
+// withWorkerAction creates the workerAction option, informing a cluster
```
```suggestion
// WithWorkerAction creates a WorkerAction option, informing a cluster
```

erikgrinaker commented on pkg/cmd/roachtest/tests/util.go:
```diff
@@ -0,0 +1,82 @@
+//
+// TODO(nvanbenschoten): this function should take a context and be responsive
+// to context cancellation.
+func WaitFor3XReplication(t test.Test, db *sql.DB) {
```
Like the name change. Should eventually have a variant that detects the expected number of replicas.

erikgrinaker reviewed (approved):
LGTM. The two last commits could arguably be merged, but 🤷‍♀️.

tbg commented:
TFTR! Incorporated your suggestion. "Unfortunately", about the time you were reviewing, I pushed another batch of commits. Mind taking a look at them too and signing off again? "roachtest: fix (another) stray newMonitor" is the first new commit I think.

tbg commented:
bors r=erikgrinaker
